src/app/hooks/ens.tsx

```diff
@@ -0,0 +1,73 @@
+import { useMemo, useCallback } from "react";
+import { useProvider } from "app/hooks";
+import { AvatarResolver } from "@ensdomains/ens-avatar";
+import { http } from "viem";
+import { mainnet } from "viem/chains";
+import { createEnsPublicClient } from "@ensdomains/ensjs";
+
+const client = createEnsPublicClient({
+  chain: mainnet,
+  transport: http(),
+});
+
+const useEns = () => {
+  const provider = useProvider();
```

Comment:
Can we avoid using this provider? Because it cause rerender for every chain switch. We need to display the same name/avatar for every chain.

Reply:
Or we can use `getClientProvider(1)` (1 - ethereum chain id)

Reply:
+